plugin/evm/imports_test.go

```
@@ -76,3 +85,120 @@ func TestMustNotImport(t *testing.T) {
+			errorMsg.WriteString(fmt.Sprintf("   Used in %d file(s):\n", len(fileList)))
+			for _, file := range fileList {
+				errorMsg.WriteString(fmt.Sprintf("   • %s\n", file))
+			}
+			errorMsg.WriteString("\n")
+		}
+
+		require.Fail(t, errorMsg.String())
+	}
+}
+
+// loadAllowedPackages reads the allowed packages from the specified file
+func loadAllowedPackages(filename string) (map[string]struct{}, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open allowed packages file: %w", err)
+	}
+	defer file.Close()
+
+	allowed := make(map[string]struct{})
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line != "" && !strings.HasPrefix(line, "#") {
+			// Remove quotes if present
+			line = strings.Trim(line, `"`)
+			allowed[line] = struct{}{}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("failed to read allowed packages file: %w", err)
+	}
+
+	return allowed, nil
+}
+
+// findFilteredLibevmImportsWithFiles finds all libevm imports in the codebase,
+// excluding underscore imports and "eth*" named imports.
+// Returns a map of import paths to the set of files that contain them
+func findFilteredLibevmImportsWithFiles(rootDir string) (map[string]map[string]struct{}, error) {
+	imports := make(map[string]map[string]struct{})
+	libevmRegex := regexp.MustCompile(`^github\.com/ava-labs/libevm/`)
+
+	return imports, filepath.Walk(rootDir, func(path string, _ os.FileInfo, err error) error {
+		if err != nil || !strings.HasSuffix(path, ".go") {
+			return err
+		}
+
+		// Skip generated files, main_test.go, and tempextrastest directory
+		filename := filepath.Base(path)
+		if strings.HasPrefix(filename, "gen_") || strings.Contains(path, "core/main_test.go") || strings.Contains(path, "tempextrastest/") {
+			return nil
+		}
+
+		node, err := parser.ParseFile(token.NewFileSet(), path, nil, parser.ParseComments)
+		if err != nil {
```

This seems bad... why not return an error?

---

that would be more concerning if there were.